modules/blockuser.cpp

```diff
@@ -173,21 +181,22 @@ class CBlockUser : public CModule {
 
 		// Disconnect all clients
 		vector<CClient*> vpClients = pUser->GetAllClients();
-		vector<CClient*>::iterator it;
-		for (it = vpClients.begin(); it != vpClients.end(); ++it) {
-			(*it)->PutStatusNotice(MESSAGE);
-			(*it)->Close(Csock::CLT_AFTERWRITE);
+		for (CClient* pClient : vpClients) {
+			pClient->PutStatusNotice("Blocked: " + (sReason.empty() ? DEFAULT_REASON : sReason));
+			pClient->Close(Csock::CLT_AFTERWRITE);
 		}
 
 		// Disconnect all networks from irc
 		vector<CIRCNetwork*> vNetworks = pUser->GetNetworks();
-		for (vector<CIRCNetwork*>::iterator it2 = vNetworks.begin(); it2 != vNetworks.end(); ++it2) {
-			(*it2)->SetIRCConnectEnabled(false);
+		for (CIRCNetwork* pNetwork : vNetworks) {
+			pNetwork->SetIRCConnectEnabled(false);
 		}
 
-		SetNV(pUser->GetUserName(), "");
+		SetNV(pUser->GetUserName(), sReason);
```

Comment:
I don't have any favorite JSON parser, you can pick one. And the format is compatible between different parsers anyway, so it can be changed to some other library in future if needed.
Format of user settings on the user's machine can't be changed so easily, that's why I'd prefer this PR to use JSON.